hoomd/md/pytest/test_thermo.py

```diff
@@ -0,0 +1,166 @@
+    np.testing.assert_allclose(thermo.kinetic_temperature, 2*thermo.kinetic_energy/thermo.degrees_of_freedom, rtol=1e-5)
+    np.testing.assert_allclose(thermo.pressure, 2.0/3*thermo.kinetic_energy/20**3, rtol=1e-5)
+    np.testing.assert_allclose(thermo.pressure_tensor, [8.0/20.0**3, 0., 0., 0., 0., 0.], rtol=1e-5)
+
+
+def test_basic_system_2d(simulation_factory, lattice_snapshot_factory):
+    group = hoomd.filter.Type(['A'])
+    groupB = hoomd.filter.Type(['B'])
+    thermo = hoomd.md.compute.ThermodynamicQuantities(group)
+    thermoB = hoomd.md.compute.ThermodynamicQuantities(groupB)
+    snap = lattice_snapshot_factory(particle_types=['A', 'B'], dimensions=2, n=2)
+    if snap.exists:
+        snap.particles.velocity[:] = [[-1, 0, 0], [2, 0, 0]]*2
+        snap.particles.typeid[:] = [0, 1, 0, 1]
+    sim = simulation_factory(snap)
+    sim.always_compute_pressure = True
+    sim.operations.add(thermo)
+    sim.operations.add(thermoB)
+
+    integrator = hoomd.md.Integrator(dt=0.0001)
+    integrator.methods.append(hoomd.md.methods.NVT(group, tau=1, kT=1))
+    integrator.methods.append(hoomd.md.methods.Langevin(groupB, kT=1, seed=3))
+    sim.operations.integrator = integrator
+
+    sim.operations.schedule()
+    sim.run(1)
+
+    # tests for group A
+    assert thermo.num_particles == 2
+    assert thermo.rotational_degrees_of_freedom == 0
+    assert thermo.translational_degrees_of_freedom == 4
+    assert thermo.degrees_of_freedom == 4
+    np.testing.assert_allclose(thermo.potential_energy, 0.0)
+    np.testing.assert_allclose(thermo.rotational_kinetic_energy, 0.0, rtol=1e-5)
+    np.testing.assert_allclose(thermo.translational_kinetic_energy, 1.0, rtol=1e-5)
+    np.testing.assert_allclose(thermo.kinetic_energy, 1.0, rtol=1e-5)
+    np.testing.assert_allclose(thermo.kinetic_temperature, 2*thermo.kinetic_energy/thermo.degrees_of_freedom, rtol=1e-5)
+    np.testing.assert_allclose(thermo.pressure, thermo.kinetic_energy/2.0**2, rtol=1e-5)
+    np.testing.assert_allclose(thermo.pressure_tensor, (2.0/2.0**2, 0., 0., 0., 0., 0.), rtol=1e-5)
+
+    # tests for group B
+    assert thermoB.num_particles == 2
+    assert thermoB.rotational_degrees_of_freedom == 0
+    assert thermoB.translational_degrees_of_freedom == 4
+    assert thermoB.degrees_of_freedom == 4
+    np.testing.assert_allclose(thermoB.potential_energy, 0.0)
+    np.testing.assert_allclose(thermoB.rotational_kinetic_energy, 0.0, rtol=1e-5)
+    np.testing.assert_allclose(thermoB.translational_kinetic_energy, 4.0, rtol=1e-3)
+    np.testing.assert_allclose(thermoB.kinetic_energy, 4.0, rtol=1e-3)
+    np.testing.assert_allclose(thermoB.kinetic_temperature, 2*thermoB.kinetic_energy/thermoB.degrees_of_freedom, rtol=1e-3)
+    np.testing.assert_allclose(thermoB.pressure, thermoB.kinetic_energy/2.0**2, rtol=1e-3)
+    np.testing.assert_allclose(thermoB.pressure_tensor, (8.0/2.0**2, 0., 0., 0., 0., 0.), rtol=1e-3, atol=1e-1)
```

Comment:
It may be worth abstracting this into a function since it is the same check with just different values. I am fine either way though.